spikeinterface/extractors/tests/test_stream_ibl.py

```diff
@@ -0,0 +1,190 @@
+
+    def test_representation(self):
+        expected_recording_representation = "StreamingIblExtractor: 385 channels - 1 segments - 30.0kHz - 5812.311s"
+        assert repr(self.recording) == expected_recording_representation
+
+    def test_dtype(self):
+        expected_datatype = "int16"
+        assert self.recording.get_dtype().name == expected_datatype
+
+    def test_channel_ids(self):
+        expected_channel_ids = [f"AP{index}" for index in range(384)] + ["SY0"]
+        self.assertListEqual(list1=list(self.recording.get_channel_ids()), list2=expected_channel_ids)
+
+    def test_gains(self):
+        expected_gains = np.ones(shape=385)
+        assert_array_equal(x=self.recording.get_channel_gains(), y=expected_gains)
+
+    def test_offsets(self):
+        expected_offsets = np.zeros(shape=385)
+        assert_array_equal(x=self.recording.get_channel_offsets(), y=expected_offsets)
+
+    def test_int16_gain_to_uV(self):
+        """A temporary property to maintain information of how to scale source int16 data to uV."""
+        expected_int16_gain_to_uV = np.concatenate((2.34375 * np.ones(shape=384), [1171.875]))
+        assert_array_equal(x=self.recording.get_property("int16_gain_to_uV"), y=expected_int16_gain_to_uV)
+        
+    def test_probe_representation(self):
+        expected_exception = ValueError
+        expected_error_message = "There is no Probe attached to this recording. Use set_probe(...) to attach one."
+        with self.assertRaisesRegex(
+            expected_exception=expected_exception,
+            expected_regex=f"^{escape(expected_error_message)}$",
+        ):
+            self.recording.get_probe()
+
+    def test_property_keys(self):
+        expected_property_keys = [
+            'gain_to_uV',
+            'offset_to_uV',
+            'shank',
+            'shank_row',
+            'shank_col',
+            'good_channel',
+            'inter_sample_shift',
+            'adc',
+            'index_on_probe',
+            'int16_gain_to_uV',
+        ]
+        self.assertCountEqual(first=self.recording.get_property_keys(), second=expected_property_keys)
+
+    def test_trace_shape(self):
+        expected_shape = (21, 385)
+        self.assertTupleEqual(tuple1=self.small_scaled_trace.shape, tuple2=expected_shape)
+
+    def test_scaled_trace_dtype(self):
+        expected_dtype = np.dtype("float32")
+        assert self.small_scaled_trace.dtype == expected_dtype
+
+    def test_unscaled_trace_dtype(self):
+        expected_dtype = np.dtype("int16")
+        assert self.small_unscaled_trace.dtype == expected_dtype
+
+
+if __name__ == '__main__':
+    TestDefaultStreamingIblExtractorApBand.setUpClass()
+    test1 = TestDefaultStreamingIblExtractorApBand()
+    test1.setUp()
+    test1.test_get_stream_names()
+    test1.test_representation()
+    test1.test_dtype()
+    test1.test_channel_ids()
```

Comment:
Also, is there a reason to avoid using `unittest.main()` for things like this?

Reply:
mainly to allow to run a subset of tests when calling the main function. Also if tests fail in this case things are much easier to debug than running through unittest (you can use the debugger and navigate the stack)